src/modules/job-list/match.c

```diff
@@ -54,6 +56,20 @@ struct timestamp_value {
     match_comparison_t t_comp;
 };
 
+#define CONSTRAINT_COMPARISON_MAX 100000
```

Comment:
This limit feels a bit low, for example:
```
--------------------------------------
> 0.00000
> job-list.list
> flags=topic,payload,route userid=unknown rolemask=none nodeid=any matchtag=1
> {"max_entries": 1000, "attrs": ["dependencies", "ntasks", "t_run", "state", "result", "t_cleanup", "duration", "name", "nodelist", "nnodes", "queue", "annotations", "userid"], "since": 0.0, "constraint": {"and": [{"hostlist": ["corona[1-1000]"]}, {"userid": [6885]}]}}
--------------------------------------
< 0.04141
< job-list.list
< flags=topic,payload,route userid=6885 rolemask=owner,local errnum=22 matchtag=1
< Excessive comparisons made, limit search via states or since
```

The limit was hit in 40ms on corona. If we say on this system the limit should prevent job-list getting stuck doing matches for more than say 200ms, then we can probably raise the limit by 4-5x.

Reply:
ahhh i set this limit before the hostlist support (which is in the follow up PR).

Maybe just upping it to like 1-2 million would even be fine.  I think the main goal is to just not DoS job-list an obscene amount.